rpcs3/rpcs3qt/vfs_dialog_path_widget.cpp

```diff
@@ -0,0 +1,108 @@
+#include "vfs_dialog_path_widget.h"
+#include "Utilities/Config.h"
+
+#include <QFileDialog>
+#include <QCoreApplication>
+#include <QHBoxLayout>
+#include <QPushButton>
+
+vfs_dialog_path_widget::vfs_dialog_path_widget(const QString& name, const QString& current_path, QString default_path, gui_save list_location, std::shared_ptr<gui_settings> guiSettings, QWidget* parent)
+	: QWidget(parent), m_default_path(std::move(default_path)), m_list_location(std::move(list_location)), m_gui_settings(std::move(guiSettings))
+{
+	m_dir_list = new QListWidget(this);
+
+	const QStringList alldirs = m_gui_settings->GetValue(m_list_location).toStringList();
+
+	QListWidgetItem* selected_item = nullptr;
+
+	for (const QString& dir : alldirs)
+	{
+		QListWidgetItem* item = new QListWidgetItem(dir, m_dir_list);
+		if (dir == current_path)
+			selected_item = item;
+	}
+
+	// We must show the currently selected config.
+	if (!selected_item)
+		selected_item = new QListWidgetItem(current_path, m_dir_list);
+
+	selected_item->setSelected(true);
+
+	m_dir_list->setMinimumWidth(m_dir_list->sizeHintForColumn(0));
+
+	QPushButton* addDir = new QPushButton(QStringLiteral("+"));
```

Comment:
This better be `add_directory_button`.

Reply:
This is all moved code that i didn't touch.
But yes, you are right